crates/iceberg-catalog/src/service/token_verification.rs

```diff
@@ -18,24 +18,107 @@ use serde::de::DeserializeOwned;
 use serde::Deserialize;
 use std::fmt::Debug;
 use std::str::FromStr;
+use std::sync::Arc;
 use url::Url;
+use uuid::Uuid;
 
 #[derive(Debug, Clone)]
 pub enum AuthDetails {
     JWT(Claims),
 }
 
+impl AuthDetails {
+    #[must_use]
+    pub fn user_id(&self) -> Option<Uuid> {
+        match self {
+            // TODO: sub is a string, we don't want to deal with that in our db, can we prescribe
+            //       uuid in the jwt?
+            // should we use oid / id? Airflow is doing that:
+            //  - google:   "username": "google_" + data.get("id", ""),
+            //  - azure:    "username": me["oid"],
+            //  - keycloak: "username": data.get("preferred_username", ""),
+            // this seems to conflate the username with the user id though
+            Self::JWT(claims) => claims.sub.parse().ok(),
```

Comment:
@c-thiel

Reply:
I don't think we can prescribe uuids - zitadel for example is not using uuids.

oid is the most correct thing we can use I believe - according to the Azure docs its a global identifier:
https://learn.microsoft.com/en-us/entra/identity-platform/id-token-claims-reference.

We need to make sure though that its also part of OAuth2 tokens and not only the identity token

Reply:
[access token docs](https://learn.microsoft.com/en-us/entra/identity-platform/access-token-claims-reference#payload-claims) also say sub is suitable as key in DB:


sub | String | The principal associated with the token. For example, the user of an application. This value is immutable, don't reassign or reuse. The subject is a pairwise identifier that's unique to a particular application ID. If a single user signs into two different applications using two different client IDs, those applications receive two different values for the subject claim. Using the two different values depends on architecture and privacy requirements. See also the oid claim, which does remain the same across applications within a tenant. | This value can be used to perform authorization checks, such as when the token is used to access a resource, and can be used as a key in database tables.
-- | -- | -- | --
oid | String, a GUID | The immutable identifier for the requestor, which is the verified identity of the user or service principal. This ID uniquely identifies the requestor across applications. Two different applications signing in the same user receive the same value in the oid claim. The oid can be used when making queries to Microsoft online services, such as the Microsoft Graph. The Microsoft Graph returns this ID as the id property for a given user account. Because the oidallows multiple applications to correlate principals, to receive this claim for users use the profile scope. If a single user exists in multiple tenants, the user contains a different object ID in each tenant. Even though the user logs into each account with the same credentials, the accounts are different. | This value can be used to perform authorization checks, such as when the token is used to access a resource, and can be used as a key in database tables.






Regarding UUID vs String, I was wondering about the data type of the column in the db, somehow this has to become our primary key, and we'll need to make it a composite of issuer & oid/sub since oid/sub won't be unique across idps, so i understand, we'd be concatenating them and make it a text? For entra-id, this'd be in the 110 bytes range per key if we assume a uuid as oid/sub.

Reply:
Decision: oid

Reply:
Decision: Concatenation with <url-encoded issuer url>/<oid> (Validate that oid contains of alphanumeric chars and -)